tests/test_alias.rs

```diff
@@ -16,6 +16,138 @@ use crate::common::TestEnvironment;
 
 pub mod common;
 
+#[test]
+fn test_alias_basic() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    test_env.add_config(
+        br#"[alias]
+    b = ["log", "-r", "@", "-T", "branches"]
+    "#,
+    );
+    test_env.jj_cmd_success(&repo_path, &["branch", "my-branch"]);
+    let stdout = test_env.jj_cmd_success(&repo_path, &["b"]);
+    insta::assert_snapshot!(stdout, @r###"
+    @ my-branch
+    ~ 
+    "###);
+}
+
+#[test]
+fn test_alias_calls_unknown_command() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    test_env.add_config(
+        br#"[alias]
+    foo = ["nonexistent"]
+    "#,
+    );
+    // Should get an error about the unknown command
+    test_env.jj_cmd_cli_error(&repo_path, &["foo"]);
+}
+
+#[test]
+fn test_alias_cannot_override_builtin() {
```

Comment:
FYI there is a way to override it, `jj -- log`.  Escaped subcommands are treated as external.  No idea why this is done but cargo has been working around this for a while.  Cargo calls the built-in command, ignoring the arguments.  See https://github.com/rust-lang/cargo/blob/master/src/bin/cargo/cli.rs#L265

There isn't a good way to handle this with derives in clap3.  This is fixed in clap v4 (clap v3 with `unstable-v4` feature flag), see clap-rs/clap#3703.